core/src/context.rs

```diff
@@ -48,41 +56,304 @@ impl Context {
+                let Some((_, package)) = self.transforms.get(&(name.clone(), output_format.clone())) else {
+                    return Err(CoreError::MissingTransform(name.clone(), output_format.to_string()));
+                };
+
+                let mut input = Pipe::new();
+                let mut output = Pipe::new();
+                write!(
+                    &mut input,
+                    "{}",
+                    self.serialize_element(from, output_format)?
+                )?;
+
+                let wasi_env = WasiState::new("")
+                    .stdin(Box::new(input))
+                    .stdout(Box::new(output.clone()))
+                    .args(["transform", name, &output_format.to_string()])
+                    .finalize(&mut self.store)?;
+
+                let import_object =
+                    wasi_env.import_object(&mut self.store, &package.wasm_module)?;
+                let instance =
+                    Instance::new(&mut self.store, &package.wasm_module, &import_object)?;
+
+                // Attach the memory export
+                let memory = instance.exports.get_memory("memory")?;
+                wasi_env
+                    .data_mut(&mut self.store)
+                    .set_memory(memory.clone());
+
+                // Call the main entry point of the program
+                let main_fn = instance.exports.get_function("_start")?;
+                main_fn.call(&mut self.store, &[])?;
+
+                // Read the output of the package from stdout
+                let result = {
+                    let mut buffer = String::new();
+                    output.read_to_string(&mut buffer)?;
+                    Self::deserialize_compound(&buffer)
+                };
+
+                result
+            }
+        }
     }
 
     /// Borrow information about a package with a given name
     pub fn get_package_info(&self, name: &str) -> Option<&PackageInfo> {
-        self.packages_by_name.get(name).map(|pkg| pkg.info.as_ref())
+        self.packages.get(name).map(|pkg| pkg.info.as_ref())
+    }
+
+    /// Serialize and element into a string that can be sent to a package
+    pub fn serialize_element(
+        &self,
+        element: &Element,
+        output_format: &OutputFormat,
+    ) -> Result<String, CoreError> {
+        let entry = self.element_to_entry(element, output_format)?;
+        serde_json::to_string_pretty(&entry).map_err(|e| e.into())
+    }
+
+    /// Deserialize a compound (i.e a list of JsonEntries) that are recived from a package
+    pub fn deserialize_compound(input: &str) -> Result<Element, CoreError> {
+        let entries: Vec<JsonEntry> = serde_json::from_str(input)?;
+
+        // Convert the parsed entries into real Elements
+        let elements: Vec<Element> = entries.into_iter().map(Self::entry_to_element).collect();
+        Ok(Element::Compound(elements))
+    }
+
+    /// Convert an JsonEntry to a Element
+    fn entry_to_element(entry: JsonEntry) -> Element {
```

Comment:
It might be worth to `impl Into` here, maybe? It will also be clearer that this is associated with `Element`

Reply:
Yeah, I wondered about that too. However,  since `element_to_entry` requires additional information it could not be implemented for that function and I thought it be best to keep the other way around as function too to avoid confusion.